padertorch/data/segment.py

```
@@ -0,0 +1,413 @@
+
+        # Discard examples that are shorter than `length`
+        if to_segment_length < self.length:
+            import lazy_dataset
+            raise lazy_dataset.FilterException()
+
+        boundaries, segmented = self.segment(to_segment, to_segment_length,
+                                             axis=axis)
+
+        segmented_examples = list()
+        if self.copy_keys is not None:
+            example = {key: example.pop(key) for key in self.copy_keys}
+        for idx, (start, stop) in enumerate(boundaries):
+            example_copy = copy(example)
+            example_copy.update({to_segment_keys[i]: value[idx]
+                                 for i, value in enumerate(segmented)})
+
+            example_copy.update(segment_start=start, segment_stop=stop)
+            segmented_examples.append(deflatten(example_copy))
+        return segmented_examples
+
+
+    def segment(self, to_segment, to_segment_length, axis):
+
+        anchor = get_anchor(
+            to_segment_length, self.length, self.shift,
+            mode=self.anchor_mode, rng=self.rng
+        )
+
+        boundaries = get_segment_boundaries(
+            to_segment_length, self.length, self.shift,
+            anchor=self.anchor_mode, rng=self.rng
+        )
+
+        segmented = [segment(signal, length=self.length, shift=self.shift,
+                       rng=self.rng, axis=axis[i], anchor=anchor)
+                     for i, signal in enumerate(to_segment)]
+        return boundaries, segmented
+
+    def get_to_segment_keys(self, example):
+        if self.include is None:
+            return [
+                key for key, value in example.items()
+                if not key in self.exclude and isinstance(value, np.ndarray)
+            ]
+        else:
+            return [
+                key for key in example.keys()
+                if not key in self.exclude and
+                any([include_key in key for include_key in self.include])
+            ]
```

What if a key is in `self.include` but not in the example? Should the code pass or should it raise an exception?

---

It should probably raise an exception, I can add that.